### ISSUE TITLE
\OC\Files\Config\MountProviderCollection::registerProvider() should add mounts for user already set up

### ISSUE THREAD

**Issue by dagan** [owncloud/core#16016]
```php
/**
 * Add a provider for mount points
 *
 * @param \OCP\Files\Config\IMountProvider $provider
 */
public function registerProvider(IMountProvider $provider) {
	$this->providers[] = $provider;
}
```
https://github.com/owncloud/core/blob/stable8/lib/private/files/config/mountprovidercollection.php#L47-L54

If ```OC\Files\Config\MountProviderCollection::registerProvider()``` is called after any user has already been set up via ```OC\Files\Filesystem::initMountPoints($user)```, that user/those users will never get the mount points that would have been returned by the provider. This is particularly problematic for the logged-in user, since any call to ```OC\Util::setupFS()```, will load mounts for the logged in user.

This has the effect of preventing any instance of ```OCP\Files\Config\IMountProvider``` from depending on ```OCP\Files\IRootFolder``` since retrieving the root folder results in a call to ```OC\Util::setupFS()```.

Perhaps an array field can be added to store each user passed to ```::getMountsForUser()``` and a call to ```::registerProvider()``` could iterate over each previously processed user, retrieving the user's mounts and adding them via ```OC\Files\Filesystem::getMountManager()->addMount()```?

**Comment by DeepDiver1975** [comment]
@icewind1991 can you please have a look? THX

**Comment by cmonteroluque** [comment]
@icewind1991 ping

**Comment by icewind1991** [comment]
Fix here: #16158
